### PR TITLE
Text annotations

### PR DESCRIPTION
**Pull request by manapeirov**
https://ag-grid.atlassian.net/browse/AG-12357
https://ag-grid.atlassian.net/browse/AG-12379
https://ag-grid.atlassian.net/browse/AG-12381

### CODE REVIEW

File: packages/ag-charts-enterprise/src/features/annotations/note/noteScene.ts
Comment (lsjroberts) [inline packages/ag-charts-enterprise/src/features/annotations/note/noteScene.ts:1]: Ignore this file. Note annotation will be in another PR.

File: packages/ag-charts-enterprise/src/features/annotations/scenes/textualStartEndScene.ts [hunk 1]
```diff
@@ -0,0 +1,264 @@
...
+        placement: 'inside',
+        spacing: 0,
+    };
+
+    public readonly handleLayout: AnchoredLayout = {
+        position: 'bottom',
+        alignment: 'left',
+        placement: 'outside',
+        spacing: 0,
+    };
```
Comment (lsjroberts) [inline packages/ag-charts-enterprise/src/features/annotations/scenes/textualStartEndScene.ts]: I think we can remove these. My approach with these didn't work. `handleLayout` is not used anywhere as far as I can see, and `labelLayout` can be replaced with just `datum.position` and `datum.alignment` as I did in `TextualPointScene`.

File: packages/ag-charts-enterprise/src/features/annotations/scenes/textualStartEndScene.ts [hunk 2]
```diff
@@ -0,0 +1,264 @@
...
+            case 'center':
+                y = bbox.y + bbox.height / 2;
+                break;
+            case 'bottom':
+                y = bbox.y + bbox.height - spacingY;
+                break;
+        }
+
+        return { x, y };
+    }
```
Comment (lsjroberts) [inline packages/ag-charts-enterprise/src/features/annotations/scenes/textualStartEndScene.ts]: We can remove this now as well.

File: packages/ag-charts-enterprise/src/features/annotations/states/textualStartEndState.ts [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+            },
+            'waiting-first-render': {
+                render: {
+                    target: 'end',
+                    action: () => {
+                        ctx.node()?.toggleActive(true);
+                        ctx.showAnnotationOptions();
+                    },
+                },
+            },
```
Comment (lsjroberts) [inline packages/ag-charts-enterprise/src/features/annotations/states/textualStartEndState.ts:75]: Maybe move this up a few lines to between `start` and `end`?

File: packages/ag-charts-enterprise/src/features/annotations/states/textualStartEndState.ts [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+                    },
+                    {
+                        guard: ({ key, shiftKey }: { key: string; shiftKey: boolean }) => !shiftKey && key === 'Enter',
+                        target: StateMachine.parent,
+                        action: onSave,
+                    },
+                    {
+                        target: 'edit',
+                        action: onSave,
+                    },
```
Comment (lsjroberts) [inline packages/ag-charts-enterprise/src/features/annotations/states/textualStartEndState.ts]: We don't want this one anymore as in `TextualPointState`, as we don't need to read the size from the label bbox. And we shouldn't save until they press enter, so the escape key works properly.

**Review by lsjroberts — CHANGES_REQUESTED** [review]

**Comment by lsjroberts** [comment]
Sonarqube is unhappy about some duplication. I'd rather get to the end of these text annotations and see what can be de-duped once we're happy with it all and confident it is stable. If that's ok with you Alan?

**Review by lsjroberts — APPROVED** [review]
LGTM, ignoring the sonarqube duplications as noted

File: packages/ag-charts-enterprise/src/features/annotations/note/noteScene.ts [hunk 5]
```diff
@@ -1,23 +1,87 @@
 import { _Scene, _Util } from 'ag-charts-community';
 
-import { AnnotationType } from '../annotationTypes';
+import { type AnnotationContext, AnnotationType } from '../annotationTypes';
+import { convertPoint } from '../annotationUtils';
 import { AnnotationScene } from '../scenes/annotationScene';
-import { TextualScene } from '../scenes/textualScene';
+// import { DivariantHandle } from '../scenes/handle';
```
Comment (alantreadway) [inline packages/ag-charts-enterprise/src/features/annotations/note/noteScene.ts]: Nit: Remove?

File: packages/ag-charts-enterprise/src/features/annotations/note/noteScene.ts
Comment (alantreadway) [inline packages/ag-charts-enterprise/src/features/annotations/note/noteScene.ts:1]: I see a lot of commented out code here, should we be including this as part of this PR?

**Review by alantreadway — COMMENTED** [review]
Generally LGTM - can we cleanup the half-done note support before merge?

**Review by alantreadway — APPROVED** [review]
LGTM, merge on green!
